src/Serializer/AbstractSerializer.php

```diff
@@ -261,17 +261,44 @@ protected function serializeValue($value)
     }
 
     /**
+     * This method is provided as a non-BC upgrade of serializeCallable,
+     * since using the callable type raises a deprecation in some cases.
+     *
+     * @param callable $callable
+     *
+     * @return string
+     */
+    protected function serializeCallableWithoutTypeHint($callable): string
+    {
+        if (!\is_callable($callable)) {
+            throw new \InvalidArgumentException('Expecting callable, got ' . \gettype($callable));
+        }
+
+        if (\is_string($callable) && !\function_exists($callable)) {
+            return $callable;
+        }
+
+        return $this->serializeCallable($callable);
+    }
+
+    /**
+     * @internal Use serializeCallableWithoutTypeHint instead (no type in argument)
```

Comment:
We cannot mark an already existing method as `@internal` or we may be tempted in the future to make changes without worrying about BC when we must instead since it was part of the public API in the past. Just leave the note instead with a link to the issue and remove the comment from the parameter instead